src/compas/datastructures/mesh/core/basemesh.py

```diff
@@ -1287,59 +1159,635 @@ def delete_face(self, fkey):
+            if not data:
+                yield key
+            else:
+                yield key, self.face_attributes(key)
+
+    def edges(self, data=False):
+        """Iterate over the edges of the mesh.
+
+        Parameters
+        ----------
+        data : bool, optional
+            Return the edge data as well as the edge vertex keys.
+
+        Yields
+        ------
+        tuple
+            The next edge as a (u, v) tuple, if ``data`` is false.
+        tuple
+            The next edge as a ((u, v), data) tuple, if ``data`` is true.
+
+        Note
+        ----
+        Mesh edges have no topological meaning. They are only used to store data.
+        Edges are not automatically created when vertices and faces are added to
+        the mesh. Instead, they are created when data is stored on them, or when
+        they are accessed using this method.
+
+        This method yields the directed edges of the mesh.
+        Unless edges were added explicitly using :meth:`add_edge` the order of
+        edges is *as they come out*. However, as long as the toplogy remains
+        unchanged, the order is consistent.
+
+        Example
+        -------
+        >>>
+        """
+        seen = set()
+        for u in self.halfedge:
+            for v in self.halfedge[u]:
+                key = u, v
+                ikey = v, u
+                if key in seen or ikey in seen:
+                    continue
+                seen.add(key)
+                seen.add(ikey)
+                if not data:
+                    yield key
+                else:
+                    yield key, self.edge_attributes(key)
+
+    # --------------------------------------------------------------------------
+    # attributes
+    # --------------------------------------------------------------------------
+
+    def update_default_vertex_attributes(self, attr_dict=None, **kwattr):
+        """Update the default vertex attributes.
+
+        Parameters
+        ----------
+        attr_dict : dict, optional
+            A dictionary of attributes with their default values.
+            Defaults to an empty ``dict``.
+        kwattr : dict
+            A dictionary compiled of remaining named arguments.
+            Defaults to an empty dict.
+
+        Note
+        ----
+        Named arguments overwrite correpsonding key-value pairs in the attribute dictionary,
+        if they exist.
+
+        """
+        if not attr_dict:
+            attr_dict = {}
+        attr_dict.update(kwattr)
+        self.default_vertex_attributes.update(attr_dict)
+
+    def vertex_attribute(self, key, name, value=None):
+        """Get or set an attribute of a vertex.
+        
+        Parameters
+        ----------
+        key : int
+            The vertex identifier.
+        name : str
+            The name of the attribute
+        value : obj, optional
+            The value of the attribute.
+        
+        Returns
+        -------
+        None
```

Comment:
Is this really how to document different return values? I would have assumed only inline comment for this, and this would be assumed to be a 3-tuple return.